cmd/edge-proxy/main.go

```diff
@@ -90,7 +96,34 @@ func main() {
 		}
 	}
 
-	server.StartHTTPTunnel(httpTunnelAddr, externalHTTPProxyURI)
+	enableHTTPSTunnel := false
+	if httpsTunnelAddr != "" || httpsTunnelTLSCert != "" || httpsTunnelTLSKey != "" {
+		if httpsTunnelAddr == "" {
+			fmt.Printf("You must also provide a tunnel address in order to enable the HTTPS tunnel.\n")
+			os.Exit(1)
+		}
+
+		if httpsTunnelTLSCert == "" {
+			fmt.Printf("You must also provide a TLS cert file in order to enable the HTTPS tunnel.\n")
+			os.Exit(1)
+		}
+
+		if httpsTunnelTLSKey == "" {
+			fmt.Printf("You must also provide a TLS key file in order to enable the HTTPS tunnel.\n")
+			os.Exit(1)
+		}
+		enableHTTPSTunnel = true
+	}
+
+	go func() {
+		server.StartHTTPTunnel(httpTunnelAddr, externalHTTPProxyURI)
+	}()
+
+	go func() {
```

Comment:
I'm wondering why this code is structured in this way.  why create a goroutine that then checks if it should have started in the first place?  Why not check the if-statement enableHTTPSTunnel before creating the goroutine?

Reply:
Moved the if statement outside of the goroutine.